src/main/java/com/spotify/reaper/service/RepairManager.java

```diff
@@ -58,24 +58,30 @@ public void resumeRunningRepairRuns(AppContext context) throws ReaperException {
     Collection<RepairRun> running =
         context.storage.getRepairRunsWithState(RepairRun.RunState.RUNNING);
     for (RepairRun repairRun : running) {
-      Collection<RepairSegment> runningSegments =
-          context.storage.getSegmentsWithState(repairRun.getId(), RepairSegment.State.RUNNING);
-      for (RepairSegment segment : runningSegments) {
-        try (JmxProxy jmxProxy = context.jmxConnectionFactory
-            .connect(segment.getCoordinatorHost())) {
-          SegmentRunner.abort(context, segment, jmxProxy);
-        } catch (ReaperException e) {
-          LOG.debug("Tried to abort repair on segment {} marked as RUNNING, but the host was down"
-                    + " (so abortion won't be needed)", segment.getId(), e);          
-          SegmentRunner.postpone(context, segment, context.storage.getRepairUnit(repairRun.getId()));
+      if(!repairRunners.containsKey(repairRun.getId())) {
+        Collection<RepairSegment> runningSegments =
+            context.storage.getSegmentsWithState(repairRun.getId(), RepairSegment.State.RUNNING);
+        for (RepairSegment segment : runningSegments) {
+          try (JmxProxy jmxProxy = context.jmxConnectionFactory
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png 'Severity: CRITICAL') Refactor this code to not nest more than 3 if/for/while/switch/try statements. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS134)